registry/api/v2/urls.go

```diff
@@ -80,8 +80,8 @@ func NewURLBuilderFromRequest(r *http.Request, relative bool) *URLBuilder {
 			// comma-separated list of hosts, to which each proxy appends the
 			// requested host. We want to grab the first from this comma-separated
 			// list.
-			hosts := strings.SplitN(forwardedHost, ",", 2)
-			host = strings.TrimSpace(hosts[0])
+			host, _, _ = strings.Cut(forwardedHost, ",")
+			host = strings.TrimSpace(host)
```

Comment:
Aside: this should probably be changed to [`textproto.TrimString`](https://pkg.go.dev/net/textproto#TrimString) at some point, and the rest of the module audited for inappropriate uses of unicode functions in HTTP-related code.

Reply:
Good one; TIL about that function (or at least, I don't recall I've used it before). Probably needs to be checked with punicode domains etc 🤔